### ISSUE TITLE
Odd behavior on User type

### ISSUE THREAD

**Issue by howdoicomputer** [rodjek/rspec-puppet#387]
```
  1) appserver should compile into a catalogue without dependency cycles
     Failure/Error: it { should compile }
       error during compilation: Parameter managehome failed on User[jetty]: Validate method failed for class managehome: undefined method `manages_homedir?' for NilClass:Class
     # ./spec/classes/appserver_spec.rb:28:in `block (2 levels) in <top (required)>'
```

I'm on Puppet 2.7 and trying to move a piece of a massive codebase over to RSpec testing but I'm stuck on the portion of my module that handles manifests. I'm on a Mac and I can't update to later versions of Puppet.

I've already set my operating system values so that it chooses a provider that supports `managehome`.

```
      :operatingsystem => 'RedHat',
      :operatingsystemmajrelease => '6',
      :osfamily => 'RedHat',
      :kernel => 'Linux'
```

I tracked down the portion of code that is throwing the error:

https://github.com/puppetlabs/puppet/blob/2.7.x/lib/puppet/type/user.rb#L306

I'm a little confused here since Puppet is huge and I'm pretty inexperienced with it. I'm trying to print debug the user type to figure out what is wrong. If I don't provide an `operatingsystem` fact then I can pry out this output:

```
The provider is User[jetty](provider=directoryservice)
the class is Puppet::Type::User::ProviderDirectoryservice
```

But those values come up `nil` if I'm using an OS.

```
The provider is
the class is NilClass
```

Looking through the code shouldn't it be grabbing a, uh, `useradd` provider in place of the `directoryservice` provider when a different operating system is specified?

**Comment by rodjek** [comment]
This should now be resolved by #498

**Comment by howdoicomputer** [comment]
oh wow what a blast from the past thanks @rodjek
